modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/components/SidebarPanelInfoView/VersionActions.tsx

```diff
@@ -15,11 +15,32 @@
 import {ClayButtonWithIcon} from '@clayui/button';
 import ClayDropDown from '@clayui/drop-down';
 import ClayIcon from '@clayui/icon';
+import {navigate} from 'frontend-js-web';
 import React from 'react';
 
-const VersionActions = ({actions}: IProps) => {
-	const handleActionClick = ({url}: {url: string}): void => {
+const availableActionMethods: IAvailableActionMethods = {
+	default: (url: string) => {
+		navigate(url);
+	},
+	expire: (url: string): void => {
```

Comment:
We can't couple to code to an action, we should define an standard mechanism to decide the type of action.

Reply:
That's a good point @cgoncas 
I was thinking about this yesterday...
Then we need to standardise actions methods between BE/FE, don't you think?

Maybe something like `useNavigate`, `useSubmitForm`, `useWhatever`?

Reply:
The type of action is fine so far, what I can not see clearly is how to define a method for each action in the FE

Reply:
About:

The type of action is fine so far, what I can not see clearly is how to define a method for each action in the FE

Why can't you use the same approach we use with the ContentDashboardActions?

Thanks!!

Reply:
> Hey @beltranrengifo
> 
> I have been creating a proposal for you here: #2972
> 
> The types will be SUBMIT_FORM and NAVIGATE so far
> 
> What di you think?

I think that's very fine 👌

Reply:
> About:
> 
> The type of action is fine so far, what I can not see clearly is how to define a method for each action in the FE
> 
> Why can't you use the same approach we use with the ContentDashboardActions?
> 
> Thanks!!

Yeah, with this info we can refactor a bit the proposal and link the action type to the method easily as we did [here](https://github.com/beltranrengifo/liferay-portal/blob/644520aebe80879b37f1fd741d3c1befc072224e/modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/transformers/ActionsComponentPropsTransformer.js#L165) I believe.

Then I wonder if the name of the action in needed anymore 🤔